lib/dry/transaction/callable.rb

```diff
@@ -31,6 +31,8 @@ def initialize(operation)
       def call(*args, &block)
         if arity.zero?
           operation.(&block)
+        elsif args.size == 1 && args.first.is_a?(Hash)
```

Comment:
Hint: I'm not sure about how to solve this in a better way and could use some help. This line is to showcase why the deprecation warning is caused here. 

When `args` is an array with a single hash element, the splat operator pulls the hash out of the array and the hash will be the first argument of the operation. If the operations is a method with keyword arguments only, this will cause the deprecation warning `dry-transaction/lib/dry/transaction/callable.rb:35: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call`. Maybe there are other cases, where the operation arity is not -1 (like with keyword arguments only.

Any idea if this is the right place to fix this issue? IMHO args is very likely to always be an array, so I conclude we must fix this here.

Any idea if this is missing some use cases, that also will lead to the same depracation warning, but are not covered by the test suite and/or the `elsif`-block?

Any idea to improve the implementation here, I dislike the `elsif`-branch-style.